pg_shard.c

```diff
@@ -329,6 +373,73 @@ PgShardPlanner(Query *query, int cursorOptions, ParamListInfo boundParams)
 
 
 /*
+ * SafeToPushDownAggregate returns true if GROUP BY clause includes the partition column.
+ * This function returns false if the total length of target list is greater than the
+ * total number of columns on the distributed table whose Oid is the second parameter
+ * of this function.
+ */
+static bool
+SafeToPushDownAggregate(Query *query, Oid distributedTableId)
+{
+	List *groupClauseList = query->groupClause;
+	List *targetList = query->targetList;
+	bool safeToPushDownAggregates = false;
+	Var *partitionColumn = PartitionColumn(distributedTableId);
+	char *finalAttributeName = get_attname(distributedTableId, list_length(targetList));
```

Comment:
This method of detection is weird. Can you just get the number of attributes in the table and use `(list_length(target_list) > numOfTableAttrs)` or something like that?